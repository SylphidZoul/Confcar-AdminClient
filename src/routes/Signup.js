import { useState } from 'react'
import { useHistory } from 'react-router-dom'
import SignupScreen from '../components/signupScreen'
import Http from '../libs/http'

const Signup = () => {
  const [isFetching, setIsFetching] = useState(false)
  const [errorMessage, setErrorMessage] = useState('')
  const history = useHistory()

  const handleSubmit = async (e) => {
    e.preventDefault()
    const employee = {
      fullname: e.target[0].value,
      dni: e.target[1].value,
      password: e.target[2].value,
      mobile: e.target[3].value,
      hourly_pay: e.target[4].value,
      newEmployee: true
    }
    setIsFetching(true)
    try {
      const newEmployee = await Http.instance.post('employees', employee)
      setIsFetching(false)
      if (newEmployee.error) return setErrorMessage(newEmployee.body)
      history.replace('/employees')
    } catch (error) {
      setIsFetching(false)
      console.log('Sin conexi√≥n')
    }
  }

  return (
    <SignupScreen
      onSubmit={handleSubmit}
      isFetching={isFetching}
      errorMessage={errorMessage}
    />
  )
}

export default Signup
