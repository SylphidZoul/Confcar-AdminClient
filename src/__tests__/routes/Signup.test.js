import { mount } from 'enzyme'
import { HashRouter } from 'react-router-dom'
import toJson from 'enzyme-to-json'
import Signup from '../../routes/Signup'

describe('Employees Tests', () => {
  let wrapper

  beforeEach(async () => {
    wrapper = mount(
      <HashRouter>
        <Signup />
      </HashRouter>
    )
    expect(wrapper.exists).toBeTruthy()
  })

  test('Validate Snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
