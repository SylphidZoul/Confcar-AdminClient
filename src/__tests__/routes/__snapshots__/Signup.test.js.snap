// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees Tests Validate Snapshot 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Signup>
      <SignupScreen
        errorMessage=""
        isFetching={false}
        onSubmit={[Function]}
      >
        <styled.div>
          <div
            className="sc-bdnylx eyedbV"
          >
            <BackButton
              back="Empleados"
            >
              <styled.button
                onClick={[Function]}
              >
                <button
                  className="sc-pNWxx jEUbzj"
                  onClick={[Function]}
                >
                  <styled.img
                    alt="arrow-image"
                    src="arrow.svg"
                  >
                    <img
                      alt="arrow-image"
                      className="sc-jrsJCI vRVpE"
                      src="arrow.svg"
                    />
                  </styled.img>
                  Volver a 
                  Empleados
                </button>
              </styled.button>
            </BackButton>
            <styled.h1>
              <h1
                className="sc-gtssRu eTNwlm"
              >
                Registrar empleados
              </h1>
            </styled.h1>
            <styled.form
              onSubmit={[Function]}
            >
              <form
                className="sc-dlnjPT eTAuCH"
                onSubmit={[Function]}
              >
                <styled.div>
                  <div
                    className="sc-jSFkmK hojLfW"
                  >
                    <styled.img
                      alt="profile-icon"
                      src="profile.png"
                    >
                      <img
                        alt="profile-icon"
                        className="sc-gKAblj fbuNLX"
                        src="profile.png"
                      />
                    </styled.img>
                  </div>
                </styled.div>
                <styled.label
                  htmlFor="name"
                >
                  <label
                    className="sc-hKFyIo hSCLnj"
                    htmlFor="name"
                  >
                    Nombre
                  </label>
                </styled.label>
                <styled.input
                  name="name"
                  type="text"
                >
                  <input
                    className="sc-eCApGN llnnzY"
                    name="name"
                    type="text"
                  />
                </styled.input>
                <styled.label
                  htmlFor="dni"
                >
                  <label
                    className="sc-hKFyIo hSCLnj"
                    htmlFor="dni"
                  >
                    D.N.I.
                  </label>
                </styled.label>
                <styled.input
                  max={99999999}
                  min={1}
                  name="dni"
                  type="number"
                >
                  <input
                    className="sc-eCApGN llnnzY"
                    max={99999999}
                    min={1}
                    name="dni"
                    type="number"
                  />
                </styled.input>
                <styled.label
                  htmlFor="password"
                >
                  <label
                    className="sc-hKFyIo hSCLnj"
                    htmlFor="password"
                  >
                    Contraseña
                  </label>
                </styled.label>
                <styled.input
                  name="password"
                  type="text"
                >
                  <input
                    className="sc-eCApGN llnnzY"
                    name="password"
                    type="text"
                  />
                </styled.input>
                <styled.label
                  htmlFor="phone"
                >
                  <label
                    className="sc-hKFyIo hSCLnj"
                    htmlFor="phone"
                  >
                    Teléfono
                  </label>
                </styled.label>
                <styled.input
                  max={9999999999}
                  min={1}
                  name="phone"
                  type="number"
                >
                  <input
                    className="sc-eCApGN llnnzY"
                    max={9999999999}
                    min={1}
                    name="phone"
                    type="number"
                  />
                </styled.input>
                <styled.label
                  htmlFor="price"
                  price={true}
                >
                  <label
                    className="sc-hKFyIo jkwmKX"
                    htmlFor="price"
                  >
                    Pago/Hora
                  </label>
                </styled.label>
                <styled.input
                  name="price"
                  type="number"
                >
                  <input
                    className="sc-eCApGN llnnzY"
                    name="price"
                    type="number"
                  />
                </styled.input>
                <styled.button>
                  <button
                    className="sc-iCoHVE kxCzDn"
                  >
                    Registrar
                  </button>
                </styled.button>
              </form>
            </styled.form>
          </div>
        </styled.div>
      </SignupScreen>
    </Signup>
  </Router>
</HashRouter>
`;
